### ISSUE TITLE
Composite Rule not working with nested/array fields

### ISSUE THREAD

**Issue by dmason30** [illuminatech/validation-composite#3]
### What steps will reproduce the problem?

Great package! I just ran into this when trying to validate request data containing array fields.

If you have a set of rules that contain dot notation fields representing array validation the CompositeRule doesn't run, see example:

```php
public function rules() {
    return [
        'enabled' => 'required|boolean',        
        'attributes.url' => ['required', new SecureUrl], // Composite seems to get skipped
        'attributes.tags.*' => ['required', new TagsAllowed], // Composite seems to get skipped
        // Wildcards can be anywhere in the key 'person.*.email'
    ];
}
```

### What is the expected result?
`CompositeRule` validation is executed.

### What do you get instead?
Validation rules seem to be skipped.

### Solution?
In the below code from the `passes` method the `[$attribute => $value]` part I think is the issue as the attribute/data structure may need to be converted to the correct array structure. 

```php
/** CompositeRule::passes... */

$validator = $this->getValidatorFactory()->make([$attribute => $value], [
    $attribute => $this->rules(),
]);
```

Laravel must handle this translation from dot notation to array somewhere in their validation code. So I assume the solution is out there. 

You can transform the dot notation field to an array using something like the below:

```php
/** CompositeRule::passes... */

$fields = [];

// Transform attribute to matching array with value
\Arr::set($fields, $attribute, $value);

$validator = $this->getValidatorFactory()->make($fields, [
    $attribute => $this->rules(),
]);
```

The `Arr::set` transforms the dot notation to an array.

### Additional info

| Q                         | A
| ------------------------- | ---
| This Package Version      | 1.0.0
| Laravel Framework Version | 5.8.*
| PHP version               | 7.1
| Operating system          | Ubuntu

**Comment by klimov-paul** [comment]
Resolved by commit 4c70069a5d4a3e2f4c67c5e76537474563fe940c
Thank you for the reporting.

**Comment by dmason30** [comment]
@klimov-paul Great thanks for the quick fix - much appreciated
